### PR TITLE
Document downsample component

### LINKED ISSUE #1938: Docs for downsample cmd
I think we need docs for command `downsample` as well.

### PR DESCRIPTION
**Pull request by daixiang0**
Fix #1938

Signed-off-by: Xiang Dai <764524258@qq.com>

### CODE REVIEW

**Comment by daixiang0** [comment]
Now it is a subcmd for thanos too which could act as a service i think. Maybe comment it as a build-in feature in Compator too?

**Comment by bwplotka** [comment]
No, it cannot run like a service. 

If you want to run as a service then  you should use compactor indeed (: 

What I am saying is that I would love to move this as a subcommand of `thanos bucket` instead.

**Comment by daixiang0** [comment]
Get it, how about add an option for Compator to make it work as a tool? If move it as subcommand of `thanos bucket`, it looks a bit repeated.

**Comment by yeya24** [comment]
I agree with @bwplotka  that it is better to become a subcommand of `thanos bucket`.

**Comment by daixiang0** [comment]
@bwplotka hi, what need i change? Move this as a subcommand of thanos bucket and move this document?

**Comment by bwplotka** [comment]
> @bwplotka hi, what need i change? Move this as a subcommand of thanos bucket and move this document?

That would be amazing TBH (: let's double-check with the team.

@squat @metalmatze @brancz @FUSAKLA @GiedriusS  @povilasv  WDYT?

**Comment by squat** [comment]
That sounds like a great idea to me:)

**Comment by povilasv** [comment]
Definetely agree :+1: I remember `thanos downsample` was confusing when I was first learning Thanos. I had questions like: is it another thanos component? Do I need to run it? etc

So I think moving this onto bucket makes more sense

**Comment by metalmatze** [comment]
Agreed. :)

**Comment by daixiang0** [comment]
@bwplotka done, mind have a look?

File: pkg/downsample/downsample.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+	}, []string{"group"})
+	m.downsampleFailures = promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_compact_downsample_failures_total",
+		Help: "Total number of failed downsampling attempts.",
+	}, []string{"group"})
+
+	return m
+}
+
+func RunDownsample(
```
Comment (bwplotka) [inline pkg/downsample/downsample.go]: hm I think this should rather land in `cmd` to be consistent with other commands

Comment (daixiang0) [reply pkg/downsample/downsample.go]: Do you mean just keep `cmd/downsample.go` and call it in bucket? Or keep a part of  `cmd/downsample.go`?

Comment (bwplotka) [reply pkg/downsample/downsample.go]: I would say part of `bucket.go`?

Comment (daixiang0) [reply pkg/downsample/downsample.go]: I have changed it, if anything wrong, please tell me.

**Comment by bwplotka** [comment]
Ah it is also kind of braking change so let's add CHANGELOG in next PR (:

**Comment by daixiang0** [comment]
Track as #2278

File: docs/components/bucket.md [hunk 2]
```diff
@@ -389,3 +392,71 @@ Flags:
       --single-run               Run replication only one time, then exit.
 
 ```
+
+### Downsample
+
+The downsample component of Thanos implements the downsample API on top of historical data in an object storage bucket. It continuously downsamples blocks in an object store bucket as a service.
+
+```bash
+$ thanos downsample \
```
Comment (belm0) [inline docs/components/bucket.md:401]: should this be `thanos bucket downsample`, matching usage below?

Comment (GiedriusS) [reply docs/components/bucket.md:401]: @belm0, feel free to submit a PR which fixes this :P

**Pull request merged**